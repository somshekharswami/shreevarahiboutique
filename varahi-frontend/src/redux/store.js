import { configureStore } from "@reduxjs/toolkit";
import cartReducer from "./slices/cartSlice";
import authReducer from "./slices/authSlice"; 

// Load cart from localStorage
const loadCart = () => {
  try {
    const saved = localStorage.getItem("cartState");
    return saved ? JSON.parse(saved) : undefined;
  } catch {
    return undefined;
  }
};

// Save cart to localStorage
const saveCart = (cartSlice) => {
  try {
    localStorage.setItem("cartState", JSON.stringify(cartSlice));
  } catch {
    /* ignore write errors */
  }
};

const preloadedState = loadCart() ? { cart: loadCart() } : undefined;

export const store = configureStore({
  reducer: {
    cart: cartReducer,
    auth: authReducer,
  },
  preloadedState,
});

store.subscribe(() => {
  saveCart(store.getState().cart); // persist only cart
});

export default store;
