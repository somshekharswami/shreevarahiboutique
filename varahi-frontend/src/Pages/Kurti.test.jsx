import React from "react";
import { render, screen, waitFor } from "@testing-library/react";
import Kurti from "./Kurti";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import store from "../redux/store";
import axios from "axios";

jest.mock("axios");

describe("Testing Kurti Page", () => {
  beforeEach(() => {
    axios.get.mockResolvedValueOnce({
      data: [
        {
          _id: "1",
          title: "Kurti A",
          category: "Kurti",
          singlePrice: { discountedPrice: 400 },
        },
        {
          _id: "2",
          title: "Lehenga B",
          category: "Lehenga",
          singlePrice: { discountedPrice: 800 },
        },
      ],
    });
  });

  test("renders only Kurti products", async () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Kurti />
        </BrowserRouter>
      </Provider>
    );

    await waitFor(() => {
      expect(screen.getByText("Kurti A")).toBeInTheDocument();
      expect(screen.queryByText("Lehenga B")).not.toBeInTheDocument();
    });
  });

  test("matches snapshot", async () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Kurti />
        </BrowserRouter>
      </Provider>
    );

    await waitFor(() => {
      expect(screen.getByText("Kurti A")).toBeInTheDocument(); // ensures Axios finished
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
