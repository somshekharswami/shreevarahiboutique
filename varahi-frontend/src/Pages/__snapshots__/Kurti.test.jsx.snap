// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Testing Kurti Page matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="min-h-screen bg-gradient-to-br from-pink-100 to-purple-100 py-12 px-4 sm:px-10"
  >
    <div
      class="max-w-7xl mx-auto"
    >
      <div
        class="text-center mb-10"
      >
        <h1
          class="text-3xl md:text-5xl font-serif font-semibold text-[#b11257] text-center mt-1"
        >
          Kurti Collection
        </h1>
        <p
          class="text-md sm:text-lg text-gray-700 font-light font-serif max-w-2xl mx-auto leading-relaxed"
        >
          Explore the elegance of our handpicked Kurti
        </p>
      </div>
      <div
        class="grid grid-cols-2 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 xl:grid-cols-5 gap-6"
      >
        <div
          class="bg-white rounded-2xl shadow-md overflow-hidden group hover:shadow-xl transition-shadow duration-300"
        >
          <a
            class="block"
            data-discover="true"
            href="/product/1"
          >
            <div
              class="relative overflow-hidden bg-gray-50"
            >
              <img
                alt="Kurti A"
                class="w-full h-48 sm:h-56 object-contain group-hover:scale-105 transition-transform duration-300 p-2"
              />
            </div>
            <div
              class="p-4"
            >
              <h2
                class="text-base sm:text-lg font-semibold text-gray-900 mb-2 line-clamp-2 group-hover:text-pink-600 transition-colors font-serif"
              >
                Kurti A
              </h2>
              <p
                class="text-gray-700 text-sm sm:text-base font-normal font-serif line-clamp-2 mb-2"
              />
              <div
                class="flex items-center justify-between mb-3"
              >
                <div
                  class="text-xl font-sans"
                >
                  <span
                    class="line-through text-pink-700 mr-2"
                  >
                    ₹N/A
                  </span>
                  <span
                    class="font-bold text-green-700"
                  >
                    ₹400
                  </span>
                </div>
              </div>
              <button
                class="w-full font-sans px-4 py-2 mt-2 bg-pink-600 hover:bg-pink-700 text-white rounded-xl shadow-md transition-all duration-200"
              >
                View Item
              </button>
            </div>
          </a>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
